### ISSUE TITLE
Disabling dataGrouping on navigator causes spike in render time

### ISSUE THREAD

**Issue by jcairney** [highcharts/highcharts#10024]
#### Expected behaviour
Disabling navigator.series.dataGrouping should not cause a significant impact to performance.

#### Actual behaviour
Disabling navigator.series.dataGrouping causes a >400% increase in render time when calling `series[0].update({data: data},true)`, comparing with a configuration of dataGrouping which I think is approximate to having turned it off.

I am working with a data set of 10,000 data points.  I am making a chart that shows only the navigator and not the main chart.  The main series is set to visible false, lineWidth 0, etc.  In navigator options, series.dataGrouping is set to enabled: false.
```
    ...
    navigator: {
      series: {
        dataGrouping: {
          enabled:false,
        }
      }
    }
    ...
```
I found that dataGrouping option made the chart render >4x slower than a chart where dataGrouping was applied to the single series for the main chart.  It was also >4x slower than trying to approximate turning dataGrouping off in the navigator with the other options available.  Here is my approximation:

```
    ...
    navigator: {
      series: {
        dataGrouping: {
          enabled:true,
          groupPixelWidth:1,
          smoothed:true
        }
      }
    }
    ...
```

I understand that having a pixelGroupWidth of 1 is not nearly the same as having dataGrouping turned off, but this is the compromise I settled on in an effort to improve the performance of this graph for my application.

I still think there is something wrong when the rendering of the navigator with dataGrouping disabled is so much slower than turning off dataGrouping in a main graph.

#### Live demo with steps to reproduce
https://jsfiddle.net/8u5q6zje/

#### Product version
https://code.highcharts.com/stock/highstock.js
I have to assume this is the latest version, currently 7.0.2

#### Affected browser(s)
Chrome, Firefox, maybe all?

**Comment by KacperMadej** [comment]
Hi @jcairney 

Thank you for contacting us about the problem.

Disabling navigator.series.dataGrouping **will** cause a significant impact to performance.
Let me explain:

dataGrouping groups points, so less actual point is getting rendered and processed if dataGrouping is enabled.

no dataGrouping => more points => longer render time

If you disable dataGrouping on main series and render the chart with 'all' rangeSelector button selected, then rendering time is the same (at least that's what I have tested out to be true for me) as in navigator with disabled dataGrouping, because the same amount of points is being processed and rendered.

More points renders slower than less points, so to avoid that you could enable dataGrouping.
Maybe you could elaborate on the issue?

**Comment by jcairney** [comment]
@KacperMadej you make a valid point.  I wasn't comparing apples to applies.  I modified my example and found that when navigator is set to select full range, disabling dataGrouping on the main chart has approximately the same impact as disabling dataGrouping on the navigator.

https://jsfiddle.net/36sk9uy1/ (See console output)

The reason I didn't notice that disabling dataGrouping had the same performance impact on a main chart as on the navigator was that when I applied it to main charts, I was able to compensate for the performance hit by turning on the boost module.  I have several charts with no navigator, the same height and width as a navigator, showing the full range of their data (10k data points), with dataGrouping disabled, and boost module on.

For clarity, I'm making an ECG Viewer with: 

[Large chart, no navigator, range of 10 seconds of heartbeat visible, scrollbar for scrolling through 1 minute]
[Zoomed out view of heartbeat, range -2min to -1min, dataGrouping disabled, boost on]
[Zoomed out view, -1min to 0min, dataGrouping disabled, boost on]
[Hidden chart with navigator showing 0min to 1min with selected range of 10s, and event handling to update Large chart with range selected here]
[Zoomed out view, +1min to +2min, dataGrouping disabled, boost on]
[Zoomed out view, +2min to +3min, dataGrouping disabled, boost on]

However the boost module seems to have no effect in speeding up rendering of just the navigator.

Perhaps that's really the issue for me - boost has no noticeable effect at improving rendering of navigator after dataGrouping is disabled.
